### DISCUSSION TITLE
Why does `LIST_UNIQUE` and `COUNT(DISTINCT)` have different output data types?

### DISCUSSION THREAD

**Discussion by csubhodeep** [duckdb/duckdb#7840] [Q&A]
For a given table that I created using the below snippet - 

```python
import pandas as pd
import duckdb

df = pd.DataFrame({"a": ['x', 'x', 'y', 'y', 'z'], "b": [1, 2, 3, 3, 1]})

duckdb.sql("SELECT * FROM df")
```
Output
```shell
┌─────────┬───────┐
│    a    │   b   │
│ varchar │ int64 │
├─────────┼───────┤
│ x       │     1 │
│ x       │     2 │
│ y       │     3 │
│ y       │     3 │
│ z       │     1 │
└─────────┴───────┘
```

I am getting two different data types as results for the below query
```python
duckdb.sql("""
SELECT 
    a,
    LIST_UNIQUE(LIST(b)) AS n_unique,
    COUNT(DISTINCT b) AS n_unique2
FROM df
GROUP BY a
""")
```
Output
```shell
┌─────────┬──────────┬───────────┐
│    a    │ n_unique │ n_unique2 │
│ varchar │  uint64  │   int64   │
├─────────┼──────────┼───────────┤
│ x       │        2 │         2 │
│ y       │        1 │         1 │
│ z       │        1 │         1 │
└─────────┴──────────┴───────────┘
```
I am using `duckdb` version **0.8.0**
I am not sure whether this behaviour qualifies as a bug or is it by design because if it's the latter then it would be interesting to know the rationale behind it. Thanks !

**Comment by l1t1** [comment]
CLI the same behavior
```
D with t as(select 'x' a, 1 b )select a,list_unique(list(b)), count(distinct b) from t group by a;
┌─────────┬──────────────────────┬───────────────────┐
│    a    │ list_unique(list(b)) │ count(DISTINCT b) │
│ varchar │        uint64        │       int64       │
├─────────┼──────────────────────┼───────────────────┤
│ x       │                    1 │                 1 │
└─────────┴──────────────────────┴───────────────────┘
```

**Comment by Tishj** [comment]
I don't think the difference is intended, they're both acceptable solutions, though I think `uint64` (`UBIGINT`) is the better solution of the two.
`COUNT`:
```c++
AggregateFunction CountStarFun::GetFunction() {
	auto fun = AggregateFunction::NullaryAggregate<int64_t, int64_t, CountStarFunction>(LogicalType::BIGINT);
	fun.name = "count_star";
	fun.null_handling = FunctionNullHandling::SPECIAL_HANDLING;
	fun.window = CountStarFunction::Window<int64_t>;
	// TODO is there a better way to set those?
	fun.serialize = CountStarSerialize;
	fun.deserialize = CountStarDeserialize;
	return fun;
}
```
LIST_UNIQUE:
```c++
ScalarFunction ListUniqueFun::GetFunction() {
	return ScalarFunction({LogicalType::LIST(LogicalType::ANY)}, LogicalType::UBIGINT, ListUniqueFunction,
	                      ListUniqueBind);
}
```

**Reply by csubhodeep** [reply]
So would this be harmonized to either of the two data types in future? Because just to put a bit of context (and I know its not a super popular use case rather a corner-ish one) - the results coming out of a query which uses functions like `LIST_UNIQUE` could be converted to other in-memory representations done by other packages (such as `pandas.DataFrame` or `pyarrow.Table` or `polars.LazyFrame` etc) and sometimes a few operations in those representations don't handle unsigned int good enough especially if it comes down to do IO with a DB engine (e.g. Oracle which does not support unsigned int as a datatype)
